From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Thu, 6 Nov 2014 10:19:49 +0100
Subject: [PATCH] [media] imx-ipuv3-csi: fix try_fmt for raw formats

fmt is NULL for .raw-only formats as those don't have .rgb or .yuv set.

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 drivers/media/platform/imx/imx-ipuv3-csi.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/drivers/media/platform/imx/imx-ipuv3-csi.c b/drivers/media/platform/imx/imx-ipuv3-csi.c
index dbc53a075208..d212f9560380 100644
--- a/drivers/media/platform/imx/imx-ipuv3-csi.c
+++ b/drivers/media/platform/imx/imx-ipuv3-csi.c
@@ -1054,12 +1054,14 @@ static int ipucsi_try_fmt(struct file *file, void *fh,
 
 	f->fmt.pix.bytesperline = f->fmt.pix.width * bytes_per_pixel;
 	f->fmt.pix.sizeimage = f->fmt.pix.bytesperline * f->fmt.pix.height;
-	if (fmt->fourcc == V4L2_PIX_FMT_YUV420 ||
-	    fmt->fourcc == V4L2_PIX_FMT_YVU420 ||
-	    fmt->fourcc == V4L2_PIX_FMT_NV12)
-		f->fmt.pix.sizeimage = f->fmt.pix.sizeimage * 3 / 2;
-	else if (fmt->fourcc == V4L2_PIX_FMT_YUV422P)
-		f->fmt.pix.sizeimage *= 2;
+	if (fmt) {
+		if (fmt->fourcc == V4L2_PIX_FMT_YUV420 ||
+		    fmt->fourcc == V4L2_PIX_FMT_YVU420 ||
+		    fmt->fourcc == V4L2_PIX_FMT_NV12)
+			f->fmt.pix.sizeimage = f->fmt.pix.sizeimage * 3 / 2;
+		else if (fmt->fourcc == V4L2_PIX_FMT_YUV422P)
+			f->fmt.pix.sizeimage *= 2;
+	}
 
 	if ((in == V4L2_FIELD_SEQ_TB && out == V4L2_FIELD_INTERLACED_TB) ||
 	    (in == V4L2_FIELD_INTERLACED_TB && out == V4L2_FIELD_SEQ_TB) ||
